src/preferences/dialog/dlgprefcolors.cpp

```diff
@@ -345,6 +358,13 @@ void DlgPrefColors::slotEditHotcuePaletteClicked() {
     openColorPaletteEditor(hotcueColorPaletteName, true);
 }
 
+void DlgPrefColors::slotKeyColorsEnabled(int i) {
+    m_bKeyColorsEnabled = static_cast<bool>(i);
+    BaseTrackTableModel::setKeyColorsEnabled(m_bKeyColorsEnabled);
+    m_pConfig->setValue(kKeyColorsEnabledConfigKey, bKeyColorsEnabled->checkState());
+    slotUpdate();
```

Comment:
This slot is called when the checkbox is clicked (changed), so the checkbox state and the config state are now identical.
No need to update the entire page now.

Reply:
Sorry for the fragmented review, I didn't take a closer look earlier.

Reply:
I did not quite get you. Did you mean that we don't need the `slotUpdate()` call because it gets called some other way when the checkbox is clicked?

Reply:
Yes, we don't need it because the checkbox state and the config state are identical after writing to the config and `setKeyColorsEnabled()`.